h/models/group_scope.py

```diff
@@ -38,7 +41,11 @@ class GroupScope(Base):
     #: http://example.com
     #: https://web.hypothes.is
     #: http://localhost:5000
-    origin = sa.Column(sa.UnicodeText, nullable=False)
+    _origin = sa.Column("origin", sa.UnicodeText, nullable=False)
+
+    @hybrid_property
```

Comment:
At some point, it may make sense to make `origin` and `path` completely private and not readable. Small steps.